src/Compiler/CodeGen/EraseUnions.fs

```diff
@@ -313,52 +343,66 @@ let rec extraTysAndInstrsForStructCtor (ilg: ILGlobals) cidx =
     | 4 -> [ ilg.typ_Int16 ], [ mkLdcInt32 0 ]
     | 5 -> [ ilg.typ_Int32 ], [ mkLdcInt32 0 ]
     | 6 -> [ ilg.typ_UInt16 ], [ mkLdcInt32 0 ]
-    | _ -> 
+    | _ ->
         let tys, instrs = extraTysAndInstrsForStructCtor ilg (cidx - 7)
         (ilg.typ_UInt32 :: tys, mkLdcInt32 0 :: instrs)
 
-let takesExtraParams (alts: IlxUnionCase[]) = 
-    alts.Length > 1 && 
-    (alts |> Array.exists (fun d -> d.FieldDefs.Length > 0) ||
-     // Check if not all lengths are distinct
-     alts |> Array.countBy (fun d -> d.FieldDefs.Length) |> Array.length <> alts.Length) 
+let takesExtraParams (alts: IlxUnionCase[]) =
+    alts.Length > 1
+    && (alts |> Array.exists (fun d -> d.FieldDefs.Length > 0)
```

Comment:
This isn't that great either, another case where the inline comment causes the code formatting to be less than ideal.